woocommerce/wc-admin
# Add CSV download to Revenue table

Phase 1 of #311.

Adds the option to download table contents as a CSV file.

![image](https://user-images.githubusercontent.com/3616980/44790412-91db0580-ab9f-11e8-99a8-a4ad9ce8ae6f.png)

**Steps to test**

- Run `npm test` to verify tests are passing.
- Once #316 is merged, the 'steps to test' from that PR will have to be followed as well. Currently, the revenue table is populated with mock data so it can be skipped.
- Go to the _Revenue_ tab under _Analytics_.
- Press _Download_ and verify that a CSV file is downloaded to the browser and its contents match the table. Also verify the file name looks like `revenue-2018-08-30.csv`.

There might be some small conflicts with #316, basically because I'm reusing (and moving) the mock data file while the other PR deletes it.

**Notes**
- A new dependency is added: `browser-filesaver`.
- This PR implements the button for the _Revenue_ table, only. It can be easily implemented in other tables with the utils in `lib/csv`.
